jcl/src/java.base/share/classes/java/lang/J9VMInternals.java

```diff
@@ -492,25 +492,30 @@ static int fastIdentityHashCode(Object anObject) {
 	
 	
 	private static String[] getClassInfoStrings(final Class<?> clazz, String classPath){
-		String classLoader;
+		String classLoaderStr = "<Bootstrap Loader>"; //$NON-NLS-1$
+		String cpResult = "<Unknown>"; //$NON-NLS-1$
 		
-		if (classPath != null) {
-			classLoader = "<Bootstrap Loader>";
-		} else {
-			classLoader = clazz.getClassLoader().toString();
-			classPath = (String)AccessController.doPrivileged(new PrivilegedAction() {
-				public Object run() {
-					String path = null;
-					try {
-						path = clazz.getProtectionDomain().getCodeSource().getLocation().toString();
-					} catch (Exception e) {
-						path = "<Unknown>";
+		if (classPath == null) {
+			ClassLoader classLoader = clazz.getClassLoader();
+			if (classLoader != null) {
+				classLoaderStr = classLoader.toString();
+				classPath = AccessController.doPrivileged(new PrivilegedAction<String>() {
+					@Override
+					public String run() {
+						String path = null;
+						try {
+							path = clazz.getProtectionDomain().getCodeSource().getLocation().toString();
```

Comment:
Would it be better to individually null check each of these items and avoid the exception on this path?  I know this only called for formatting a NoSuchMethod exception message but it seems wasteful to throw an NPE when we could avoid it

Reply:
The waste is in proportion to the frequency with which it occurs. If it happens only vary rarely, then I'm fine with this the way it is.

Reply:
With the fix I've made to check for the bootstrap loader, this shouldn't normally throw any exception. It would only occur for a custom classloader that doesn't bother to implement protection domains or code source.

Reply:
In terms of frequency, as Dan mentioned this is only used for throwing a NoSuchMethod, which ideally would never occur.

Reply:
Although given the issue, apparently the DDR tooling throws it a number of times.

Reply:
Since some things like DDR could throw NoSuchMethodException a number of times, and we don't know if a customer classloader will implement class locations, I've made the suggested change.